sonarqube/httpHelpers.go

```diff
@@ -57,3 +60,13 @@ func httpRequestHelper(client *retryablehttp.Client, method string, sonarqubeURL
 
 	return *resp, nil
 }
+
+// https://github.com/jdamata/terraform-provider-sonarqube/issues/201
+// go-retryablehttp error contains the token/user:pass in plaintext.
+// We want to censor that secret before logging the error
+func censorError(err error, secret string) error {
```

Comment:
There are more types of sensitive data than just the token used to connect to sonar - for example what if there is an error in creating a user?  Or simply a debug statement logging the params - Would we be emitting the user details, including the password into the logs?

I wouldn't necessarily "limit" this to errors (which you do by the name of the function) and perhaps consider how would this be used if it is a user request that fails - would you call this function multiple times, once for the API token and again to mask the password?   Or pass in an array of sensitive data strings so you can make one call and sensor both?

I think ensuring that all logs are masked is something to consider, perhaps for a later issue - but here ensuring that all errors are masked (including any supplied password) should be the goal of this PR.